src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -608,12 +608,20 @@ def list_model_versions(
             filter_string=filter_string,
         )
         # Cast the MLflow model versions to the ZenML model version class.
-        model_versions = [
-            self._cast_mlflow_version_to_model_version(
-                mlflow_model_version=mlflow_model_version,
-            )
-            for mlflow_model_version in mlflow_model_versions
-        ]
+        model_versions = []
+        for mlflow_model_version in mlflow_model_versions:
+            try:
+                model_versions.append(
+                    self._cast_mlflow_version_to_model_version(
+                        mlflow_model_version=mlflow_model_version,
+                    )
+                )
+            except (AttributeError, OSError):
+                # Sometimes, the Model Registry in MlFlow can become unusable
+                # due to failed version registration or misuse. In such rare
+                # cases, it's best to suppress those versions that are not usable.
+                pass
```

Comment:
silent error handling is a bit dangerous, probably we should log some warning here at least

```suggestion
            except (AttributeError, OSError) as e:
                # Sometimes, the Model Registry in MLflow can become unusable
                # due to failed version registration or misuse. In such rare
                # cases, it's best to suppress those versions that are not usable.
                logger.warning(
                    f"Error encountered while loading MLflow model version: {e}"
                )
```